apps/agent/app/src/lib/token-utils.js

```diff
@@ -75,3 +99,50 @@ export async function getTokenName(app, address) {
 
   return tokenName || null
 }
+
+export function getPresetTokens(networkType) {
+  return PRESET_TOKENS.get(networkType) || [ETHER_TOKEN_FAKE_ADDRESS]
+}
+
+export function findTransfersFromReceipt(receipt) {
+  const { logs = [] } = receipt
+  const transferLogs = logs.filter(
+    ({ topics = [] }) => topics[0] === TOKEN_TRANSFER_EVENT_SIGNATURE
+  )
+  return transferLogs
+    .map(({ address, data, topics }) => {
+      try {
+        return {
+          token: address,
+          returnData: abi.decodeLog(TOKEN_TRANSFER_EVENT_INPUTS, data, topics),
+        }
+      } catch (_) {
+        return null
+      }
+    })
+    .filter(Boolean)
+}
+
+/**
+ * Finds the target address from a receipt, given the proxyAddress of the agent.
+ * Note that this targetAddress will only be accurate if there are no token transfers,
+ * meaning that this was an execution of some kind. In this case, the 3rd topic contained
+ * in the obtained log will be the target address.
+ * @param {object} receipt - Transaction receipt.
+ * @param {string} proxyAddress - Agent proxy address.
+ * @returns {string} - the target contract address.
+ */
+export function findTargetFromReceipt(receipt, proxyAddress) {
```

Comment:
I think we are only looking for `execute()` or `safeExecute()` invocations here, so this can be a bit smarter (it can filter and decode similarly to transfer events).

Reply:
Working on this!